packages/material-ui/src/Button/Button.js

```diff
@@ -278,6 +279,22 @@ const Button = React.forwardRef(function Button(props, ref) {
     ...other
   } = props;
 
+  const themeVariantsClasses = useThemeVariants(
+    {
+      color,
```

Comment:
This is very brittle to maintain. Do we have any automated test that would verify that a newly added prop would be passed to `useThemeVariants` with its default value?

Not important for this PR. I already have an idea for a lint rule and since it's a hook it should be pretty straight forward to track.

Reply:
Lint rule would be perfect in my opinion 👍 There isn't automated tests for this, but I am open for ideas on this one. Although if we have lint rules, won't be necessary..

Reply:
Yeah lint rule looks promising.

Reply:
https://astexplorer.net/#/gist/d094e92b26f3c9217cd78fa62cbd7306/02a10fbd4af430267ab603c7400303b8cda068dc should cover most cases. Should resolve https://github.com/mui-org/material-ui/pull/21648#discussion_r460795863 first before we finalize the lint rule.